### PR TITLE
Cluster Autoscaler with Metrics-server e2e

### PR DESCRIPTION
**Pull request by junshun**
Adding a simple flow to test cluster autoscaler and metrics-server packages. This test 
- Deploys both cluster autoscaler and metrics-server packages. 
- Uses a HorizontalPodAutoscaler on a sample busybox workload. 
- Verifies that the machine deployments scale up to meet the new demand.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/framework/cluster.go [hunk 1]
```diff
@@ -1406,3 +1409,135 @@ func (e *ClusterE2ETest) VerifyPackageControllerNotInstalled() {
...
+func (e *ClusterE2ETest) CombinedAutoScalerMetricServerTest(autoscalerName string, metricServerName string, targetNamespace string, mgmtCluster *types.Cluster) {
+	ctx := context.Background()
+	ns := "default"
+	name := "hpa-busybox-test"
+	machineDeploymentName := e.ClusterName + "-" + "md-0"
+
+	e.VerifyMetricServerPackageInstalled(metricServerName, targetNamespace, mgmtCluster)
+	e.VerifyAutoScalerPackageInstalled(autoscalerName, targetNamespace, mgmtCluster)
+
+	fmt.Printf("Metrics Server and Cluster Autoscaler ready\n")
```
Comment (jonathanmeier5) [inline test/framework/cluster.go:1501]: Should these be `e.T.Log`?

File: test/framework/cluster.go [hunk 2]
```diff
@@ -1406,3 +1409,135 @@ func (e *ClusterE2ETest) VerifyPackageControllerNotInstalled() {
...
+
+	err := e.KubectlClient.ApplyKubeSpecFromBytes(ctx, mgmtCluster, hpaBusybox)
+	if err != nil {
+		e.T.Fatalf("Failed to apply hpa busybox load %s", err)
+	}
+
+	fmt.Printf("Deploying test workload\n")
```
Comment (jonathanmeier5) [inline test/framework/cluster.go:1508]: Same?

File: test/framework/cluster.go [hunk 3]
```diff
@@ -1406,3 +1409,135 @@ func (e *ClusterE2ETest) VerifyPackageControllerNotInstalled() {
...
+		e.T.Fatalf("Failed waiting for test workload deployent %s", err)
+	}
+
+	params := []string{"autoscale", "deployment", name, "--cpu-percent=50", "--min=1", "--max=20", "--kubeconfig", e.kubeconfigFilePath()}
+	_, err = e.KubectlClient.ExecuteCommand(ctx, params...)
+	if err != nil {
+		e.T.Fatalf("Failed to autoscale deployent: %s", err)
+	}
+
+	fmt.Printf("Waiting for machinedeployment to begin scaling up\n")
```
Comment (jonathanmeier5) [inline test/framework/cluster.go:1522]: Same?

File: test/framework/cluster.go [hunk 4]
```diff
@@ -1406,3 +1409,135 @@ func (e *ClusterE2ETest) VerifyPackageControllerNotInstalled() {
...
+		e.T.Fatalf("Failed to get Running phase for machinedeployment: %s", err)
+	}
+
+	err = e.KubectlClient.WaitForMachineDeploymentReady(ctx, mgmtCluster, "2m",
+		machineDeploymentName)
+	if err != nil {
+		e.T.Fatalf("Machine deployment stuck in scaling up: %s", err)
+	}
+
+	fmt.Printf("Finished scaling up machines\n")
```
Comment (jonathanmeier5) [inline test/framework/cluster.go:1542]: Same?

**Review by jonathanmeier5 — COMMENTED** [review]
Looks great, just one nit about removing some print statements. Ready to approve when those are removed, assuming you agree they should be replaced by log calls.

File: test/framework/cluster.go (hunk 4 above)
Comment (junshun) [reply test/framework/cluster.go]: addressed

File: test/framework/cluster.go [hunk 5]
```diff
@@ -1406,3 +1409,135 @@ func (e *ClusterE2ETest) VerifyPackageControllerNotInstalled() {
...
+	}
+
+	fmt.Printf("Waiting for machinedeployment to begin scaling up\n")
+	err = e.KubectlClient.WaitJSONPathLoop(ctx, mgmtCluster.KubeconfigFile, "5m", "status.phase", "ScalingUp",
+		fmt.Sprintf("machinedeployments.cluster.x-k8s.io/%s", machineDeploymentName), constants.EksaSystemNamespace)
+	if err != nil {
+		e.T.Fatalf("Failed to get ScalingUp phase for machinedeployment: %s", err)
+	}
+
+	fmt.Printf("Waiting for machinedeployment to finish scaling up\n")
```
[the remark this answers is not in the document]
Comment (junshun) [reply test/framework/cluster.go]: done

File: test/framework/cluster.go (hunk 3 above)
Comment (junshun) [reply test/framework/cluster.go]: done

File: test/framework/cluster.go (hunk 2 above)
Comment (junshun) [reply test/framework/cluster.go]: done

File: test/framework/cluster.go (hunk 1 above)
Comment (junshun) [reply test/framework/cluster.go]: done

**Pull request merged**